cosmo/monitors/nuv_dark_monitor.py

```diff
@@ -59,19 +64,33 @@ def calculate_dark_rate(self, dataframe, xlim, ylim):
 
         counts = np.histogram(dataframe["TIME"], bins=time_bins)[0]
         npix = float((xlim[1] - xlim[0]) * (ylim[1] - ylim[0]))
-        dark_rate = np.median(counts / npix / timestep)
-        # what is going on with the whole histogram, ok to use median?
-
-        return dark_rate
+        dark_rate_array = counts / npix / timestep
+        # save the whole histogram in time bins, and then plot each of them
+
+        # make a decimal year array corresponding to the time bins of the
+        # dark rates
+        # do this with the expstart (mjd) and time array from the timeline
+        # extension
+        mjd_conversion_factor = 1.15741e-5
+        # taking the expstart, binning the time array by the timestep,
+        # removing the last element in the array (bin by front edge),
+        # and then multiplying by the conversion factor
+        mjd_array = dataframe["EXPSTART"] + dataframe["TIME_3"][::timestep][
+                                            :-1] * mjd_conversion_factor
+        mjd_array = Time(mjd_array, format='mjd')
```

Comment:
```suggestion
        mjd_array = absolute_time(expstart=dataframe['EXPSTART'], time=dataframe['TIME_3'][::timestep][:-1]).mjd  # not sure if you need the "mjd" here or not
```

To do this, you'd need to import `absolute_time` from `monitor_helpers`. This is assuming that you want a time array relative to EXPSTART, and assuming that "TIME_3" units are in seconds

Reply:
o sweet i didn't know that helper function existed!!